sdk/avalon_sdk/work_order/work_order_params.py

```diff
@@ -15,57 +15,78 @@
 import json
 import logging
 
+import schema_validation.validate as WOcheck
 import avalon_crypto_utils.crypto_utility as crypto_utility
 import avalon_crypto_utils.signature as signature
-from utility.jrpc_utility import create_error_response
-from error_code.error_status import JRPCErrorCodes
+from error_code.error_status import WorkOrderStatus
+import utility.jrpc_utility as util
 logger = logging.getLogger(__name__)
 
 
 class WorkOrderParams():
-    def __init__(
+    def __init__(self):
+        self.params_obj = {}
+
+    def create_request(
             self, work_order_id, worker_id, workload_id,
             requester_id, session_key, session_iv,
             requester_nonce, verifying_key=None, payload_format="JSON-RPC",
             response_timeout_msecs=6000, result_uri=None,
             notify_uri=None, worker_encryption_key=None,
-            data_encryption_algorithm=None):
-
-        self.params_obj = {}
+            data_encryption_algorithm=None, encrypted_session_key=None):
+        """validate and creates workorder request with received values"""
         self.set_work_order_id(work_order_id)
         self.set_response_timeout_msecs(response_timeout_msecs)
         self.set_payload_format(payload_format)
+        self.set_requester_nonce(requester_nonce)
+        self.session_key = session_key
+        self.set_worker_id(worker_id)
+        self.set_workload_id(workload_id)
+        self.set_requester_id(requester_id)
+        self.set_session_key_iv(
+            crypto_utility.byte_array_to_hex(session_iv))
         if result_uri:
             self.set_result_uri(result_uri)
         if notify_uri:
             self.set_notify_uri(notify_uri)
-        self.set_worker_id(worker_id)
-        self.set_workload_id(workload_id)
-        self.set_requester_id(requester_id)
         if worker_encryption_key:
-                self.set_worker_encryption_key(
-                    worker_encryption_key)
+            self.set_worker_encryption_key(
+                worker_encryption_key)
         if data_encryption_algorithm:
             self.set_data_encryption_algorithm(data_encryption_algorithm)
-        if worker_encryption_key and session_key:
+
+        self.set_encrypted_session_key(encrypted_session_key)
+
+        code, err = WOcheck.schema_validation(
+            "sdk_WorkOrderSubmit", self.params_obj)
+        if not code:
+            del self.params_obj
```

Comment:
Could you please add a comment why we have to explicitly delete the object here instead of relying on Garbage collector to delete this object at some point.

Reply:
done

Reply:
Can we set `self.params_obj = None` and let the garbage collector do the cleanup

Reply:
done